server/connectors/traces/haystack/tracesConnector.js

```diff
@@ -79,10 +79,22 @@ connector.getSearchableKeys = () => {
     return fieldNameFetcher
         .fetch(request)
         .then((result) => {
-            const keys = result.getNamesList();
+            const fieldNamesWithMetadata = {};
+            const names = result.getNamesList();
+            const metadata = result.getFieldMetadataList();
+
+            // create map with key as whitelisted field name
+            names.forEach((name, index) => {
+                fieldNamesWithMetadata[name] = {isRangeQuery: metadata[index] && metadata[index].getIsrangequery()};
```

Comment:
will this throw ArrayOutOfBounds exception?
Expect the scenario where someone runs this UI version with reader that doesn't support FieldMetadata.
Expect FieldMetadataList to be empty.

Reply:
Good catch - it wouldn't throw an exception, but with an empty response it would set `isRangeQuery` as undefined, rather than false.